apps/desktop/src/utils/error-handling.ts

```
@@ -0,0 +1,336 @@
+            
+            if (now > resetTime) {
+              yield* Ref.set(stateRef, { _tag: "HalfOpen" })
+            } else {
+              return yield* Effect.fail({
+                _tag: "OpenAgentsError" as const,
+                message: "Circuit breaker is open",
+                context: { state: currentState },
+                timestamp: new Date()
+              })
+            }
+          }
```

Biome (2.1.2)

[error] This case is falling through to the next case.

Add a `break` or `return` statement to the end of this case to prevent fallthrough.

(lint/suspicious/noFallthroughSwitchClause)